apix-rest-v1/src/main/java/eu/xenit/apix/rest/v1/nodes/NodesWebscript1.java

```diff
@@ -651,6 +657,88 @@ public void setParent(@UriVariable final String space, @UriVariable final String
+        final NodeRef target = new NodeRef(space, store, guid);
+        if (nodeService.exists(target)) {
+            if(permissionService.hasPermission(target, PermissionService.READ)) {
+                Conversation comments = commentService.getComments(target, skipcount, pagesize);
+                boolean canCreate = permissionService.hasPermission(target, PermissionService.CREATE_CHILDREN);
+                comments.setCreatable(canCreate);
+                writeJsonResponse(response, comments);
+            } else {
+                throw new AccessDeniedException("User does not have permission to read parent node");
+            }
+        } else {
+            writeNotFoundResponse(response, target);
+        }
+    }
+
+    @ApiOperation(value = "Appends a new comment to the given node.")
+    @Uri(value = "/nodes/{space}/{store}/{guid}/comments", method = HttpMethod.POST)
+    @ApiResponses({
+            @ApiResponse(code = 200, message = "Success"),
+            @ApiResponse(code = 403, message = "Not Authorized"),
+            @ApiResponse(code = 404, message = "Not Found")
+    })
+    public void postComment(@UriVariable String space, @UriVariable String store, @UriVariable String guid,
+            final Comment newComment, WebScriptRequest request, WebScriptResponse response) throws IOException {
+        final NodeRef target = new NodeRef(space, store, guid);
+        if (nodeService.exists(target)) {
+            Comment responseComment = commentService.addNewComment(target, newComment.getContent());
+            writeJsonResponse(response, responseComment);
+        } else {
+            writeNotFoundResponse(response, target);
+        }
+    }
+
+    @ApiOperation(value = "Updates a given comment on the given node.")
+    @Uri(value = "/nodes/{space}/{store}/{guid}/comment",
+            method = HttpMethod.PUT)
+    @ApiResponses({
+            @ApiResponse(code = 200, message = "Success"),
+            @ApiResponse(code = 403, message = "Not Authorized"),
+            @ApiResponse(code = 404, message = "Not Found")
+    })
+    public void updateComment(@UriVariable String space, @UriVariable String store, @UriVariable String guid,
+            final Comment newComment, WebScriptRequest request, WebScriptResponse response) throws IOException {
+        final NodeRef targetComment = new NodeRef(space, store, guid);
+        if (nodeService.exists(targetComment)) {
+            Comment updatedComment = commentService.updateComment(targetComment, newComment.getContent());
+            response.setStatus(200);
+            writeJsonResponse(response, updatedComment);
+        } else {
+            writeNotFoundResponse(response, targetComment);
+        }
+    }
+
+    @ApiOperation(value = "Deletes a given comment on the given node.")
+    @Uri(value = "/nodes/{space}/{store}/{guid}/comment",
+            method = HttpMethod.DELETE)
+    @ApiResponses({
+            @ApiResponse(code = 204, message = "Success"),
```

Comment:
**204** should not be used. Instead use one of https://github.com/xenit-eu/alfred-api/blob/master/developer-documentation/developer.md#2xx-success

Reply:
Idem line 735

Reply:
200 then.